mgmt/src/models/external/overlay/vpcpeering.rs

```diff
@@ -106,9 +291,43 @@ impl VpcPeeringTable {
         self.0.is_empty()
     }
 
+    fn validate_peering_addition(&self, new_peering: &VpcPeering) -> ConfigResult {
+        // Check that exposes in the new peering do not collide with any of the exposes in the
+        // existing peerings in the table, for the same VPCs
+        for vpc in [&new_peering.left.name, &new_peering.right.name] {
+            let (new_local, _) = new_peering.get_peering_manifests(vpc);
+            for peering in self.peerings_vpc(vpc) {
+                let (local, _) = peering.get_peering_manifests(vpc);
+                for new_expose in &new_local.exposes {
+                    for expose in &local.exposes {
+                        validate_overlapping(
+                            &new_expose.ips,
+                            &new_expose.nots,
+                            &expose.ips,
+                            &expose.nots,
+                        )?;
+                        validate_overlapping(
+                            &new_expose.as_range,
+                            &new_expose.not_as,
+                            &expose.as_range,
+                            &expose.not_as,
+                        )?;
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
     /// Add a [`VpcPeering`] to a [`VpcPeeringTable`]
     pub fn add(&mut self, peering: VpcPeering) -> ConfigResult {
         peering.validate()?;
+        self.validate_peering_addition(&peering)?;
```

Comment:
The same comments apply here for the two validations: unless the input config would be glaringly wrong (e.g. with a duplicate peering id name, which would not a allow a later validation -- in this case because a peering would replace another one) , I would defer it and concentrate it in a single place. This can sound inefficient, but it actually isn't necessarily:  validate_peering_addition() method seems to require the whole set of peerings so far. If you would be validating once having added all of the peerings that would be ~ O(n²) ?. But, the current code has the same complexity.

Reply:
Agree, I'll change, just like for `add_expose()`.

Agree on complexity as well, in both cases we end up comparing the same number of prefixes, so there shouldn't be any particular impact.